secure/storage/src/vault.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{Error, Permissions, Storage, Value};
+use libra_vault_client::Client;
+
+/// VaultStorage utilizes Vault for maintaining encrypted, authenticated data for Libra. This
+/// version currently matches the behavior of OnDiskStorage and InMemoryStorage. In the future,
+/// Vault will be able to create keys, sign messages, and handle permissions across different
+/// services. The specific vault service leveraged herein is called KV (Key Value) Secrets Engine -
+/// Version 2 (https://www.vaultproject.io/api/secret/kv/kv-v2.html). So while Libra Secure Storage
+/// calls pointers to data keys, Vault has actually a secret that contains multiple key value
+/// pairs.
+pub struct VaultStorage {
+    pub client: Client,
+}
+
+impl VaultStorage {
+    pub fn new(host: String, token: String) -> Self {
+        Self {
+            client: Client::new(host, token),
+        }
+    }
+
+    /// Erase all secrets and keys from the vault storage. Use with caution.
+    pub fn reset(&self) -> Result<(), Error> {
+        let secrets = self.client.list_secrets("")?;
```

Comment:
This looks dangerous, I think this should be scoped in someway (e.g. secrets with a configured prefix).

Reply:
That's an interesting point, my preference at this point would be to make an issue tracking remaining vault issues / deployment constraints. In practice, I believe for V1 most folks will use an application specific vault instance. I can see there being value, however, in namespacing as you suggest and making that accessible via configuration.